### PR TITLE
chore: use ephemeral workspaces in tests

### PR DESCRIPTION
**Pull request by mitchnielsen**
## Summary

Related to https://github.com/PrefectHQ/terraform-provider-prefect/issues/218

## To do
- [x] Add helpers for ephemeral workspaces
- [x] Implement helpers for ephemeral workspaces where workspaces are referenced -> implemented for some, let's start there before the PR gets too big and we can add more in future PR(s)

### CODE REVIEW

**Comment by mitchnielsen** [comment]
@parkedwards - I got the ball rolling here and will pick it back up after the move 👍🏼

**Comment by parkedwards** [comment]
awesome - thank you @mitchnielsen

File: internal/testutils/provider.go [hunk 1]
```diff
@@ -72,3 +79,19 @@ func NewTestClient() (api.PrefectClient, error) {
 
 	return prefectClient, nil
 }
+
+func NewRandomPrefixedString() string {
+	return TestAccPrefix + acctest.RandStringFromCharSet(RandomStringLength, acctest.CharSetAlphaNum)
+}
```
Comment (parkedwards) [inline internal/testutils/provider.go:85]: 🚀

**Comment by mitchnielsen** [comment]
@parkedwards - after some thinking, I wonder if it's nicer to just merge this one before it gets too big. Instead of doing it all in one big change, we can use this one to add the new helpers and implement it in a few places, and then I'll follow up with further implementation in separate PR(s). If you disagree just lmk and I'm happy to continue with making the changes here 👍🏼

**Review by parkedwards — APPROVED** [review]
thanks for getting the ball rolling on this dude, i know it can be a little tedious. but this is already a much needed improvement to the test suite

**Comment by parkedwards** [comment]
> @parkedwards - after some thinking, I wonder if it's nicer to just merge this one before it gets too big. Instead of doing it all in one big change, we can use this one to add the new helpers and implement it in a few places, and then I'll follow up with further implementation in separate PR(s). If you disagree just lmk and I'm happy to continue with making the changes here 👍🏼

totally reasonable to me - we could also do something like "if you touch some code, update the tests"

**Comment by mitchnielsen** [comment]
No worries at all, it's helped me get up to speed on how all of this works together 👍🏼

**Pull request merged**